### PR TITLE
Create manual-publish.yml

### PR DESCRIPTION
**Pull request by Everest2020**
Try to use this action to publish Nuget package manually.

### CODE REVIEW

**Comment by jryckman** [comment]
I would like @alejandro5042 to look at this.

**Comment by Everest2020** [comment]
I would like to know whether it works or not before applying GitVersion.

We can delete the hardcoded version number in .nuspec, and use the switch to specify the version
```
-p:NuspecProperties="PackageVersion=${{ steps.gitversion.outputs.nuGetVersionV2 }}-${{ steps.gitversion.outputs.CommitsSinceVersionSourcePadded }}"
```

**Comment by jryckman** [comment]
I'm okay with testing things out first.

**Review by jryckman — APPROVED** [review]

File: .github/workflows/manual-publish.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        name: Checkout code
+      
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+        env:
+          NUGET_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (alejandro5042) [inline .github/workflows/manual-publish.yml:29]: What is this token for?

- Nuget?
- GitHub?

There is a naming conflict here.

File: .github/workflows/manual-publish.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: dotnet restore NationalInstruments.Analyzers.sln
+     
+      - name: Build App
+        run: dotnet build NationalInstruments.Analyzers.sln --configuration release
+        
+      - name: Run Tests
+        run: dotnet test NationalInstruments.Analyzers.sln
+                
+      - name: Pack Nuget Package
+        run: dotnet pack .\src\NationalInstruments.Analyzers\NationalInstruments.Analyzers.csproj -p:NuspecFile=../../build/NI.CSharp.Analyzers.nuspec --no-build
```
Comment (alejandro5042) [inline .github/workflows/manual-publish.yml:41]: What's the version of the nuget that we pack? It should auto-increment right?

File: .github/workflows/manual-publish.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# This is a basic workflow that is manually triggered
+
+name: Manual Publish Nuget
+
+# Controls when the action will run. Workflow runs when manually triggered using the UI
+# or API.
+on:
+  workflow_dispatch:
+    # Inputs the workflow accepts.
+    inputs:
```
Comment (alejandro5042) [inline .github/workflows/manual-publish.yml]: Curious: Is this empty array necessary?

File: .github/workflows/manual-publish.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        name: Checkout code
+      
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
```
Comment (alejandro5042) [inline .github/workflows/manual-publish.yml:26]: We should get this version by a `global.json` file in the repo. I think the action should be able to read the file automatically to gain the version.

**Review by alejandro5042 — CHANGES_REQUESTED** [review]

File: .github/workflows/manual-publish.yml (hunk 1 above)
Comment (Everest2020) [reply .github/workflows/manual-publish.yml:29]: For GitHub Package.
see the reference, 
https://stackoverflow.com/questions/57889719/how-to-push-nuget-package-in-github-actions

File: .github/workflows/manual-publish.yml (hunk 2 above)
Comment (Everest2020) [reply .github/workflows/manual-publish.yml:41]: Not for this PR, already mentioned above. The version stuff will be handled in later PRs.

File: .github/workflows/manual-publish.yml (hunk 3 above)
Comment (Everest2020) [reply .github/workflows/manual-publish.yml]: I can try to remove this.

File: .github/workflows/manual-publish.yml (hunk 4 above)
Comment (Everest2020) [reply .github/workflows/manual-publish.yml:26]: IMHO, it is not necessary to update the .NET Core version very frequently, .NET Core 3.1 is LTS.

see https://dotnet.microsoft.com/download/dotnet-core

File: .github/workflows/manual-publish.yml (hunk 2 above)
Comment (Everest2020) [reply .github/workflows/manual-publish.yml:41]: Be aware that this is not the final version, I have to use push->check->push... method to verify whether this action works or not. We can make all the improvements in the final PR.

**Pull request merged**